File: javascript/leetcode/license-key-formatting.js
/**
 * @param {string} S
 * @param {number} K
 * @return {string}
 */
var licenseKeyFormatting = function(S, K) {
    res = ''
    ct = 0
    for (var i = S.length - 1; i >= 0; i--) {
        if (S[i] == '-') { continue }
        if (ct == K) {
            res = '-' + res
            ct = 0
        }
        res = S[i].toUpperCase() + res
        ct++
    }
    return res
};

File: java/leetcode/license-key-formatting.java
class Solution {
    public String licenseKeyFormatting(String S, int K) {
        StringBuilder res = new StringBuilder();
        int ct = 0;
        for (int i = S.length() - 1; i >= 0; i--) {
            if (S.charAt(i) == '-') { continue; }
            if (ct == K) {
                res.append("-");
                ct = 0;
            }
            res.append(Character.toUpperCase(S.charAt(i)));
            ct++;
        }
        return res.reverse().toString();
    }
}